src/main/java/com/ecore/roles/exception/InvalidArgumentException.java

```diff
@@ -1,10 +1,8 @@
 package com.ecore.roles.exception;
 
-import static java.lang.String.format;
-
 public class InvalidArgumentException extends RuntimeException {
 
-    public <T> InvalidArgumentException(Class<T> resource) {
-        super(format("Invalid '%s' object", resource.getSimpleName()));
+    public InvalidArgumentException(String message) {
```

Comment:
I think this exception is designed to show business validation, so it only receives the error message as an argument